android/src/main/kotlin/com/example/user_location_plugin/UserLocationPlugin.kt

```diff
@@ -0,0 +1,131 @@
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry
+
+class UserLocationPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+    PluginRegistry.ActivityResultListener, PluginRegistry.RequestPermissionsResultListener {
+
+    private lateinit var channel: MethodChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    private var permissionEventChannel: EventChannel? = null
+    private var permissionEventSource: EventChannel.EventSink? = null
+    private var permissionStreamHandler: EventChannel.StreamHandler =
+        object : EventChannel.StreamHandler {
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+                permissionEventSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+                permissionEventSource = null
+            }
+        }
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "user_location_plugin")
+        channel.setMethodCallHandler(this)
+        permissionEventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, "permission_event_channel")
+        permissionEventChannel?.setStreamHandler(permissionStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "getPlatformVersion" -> getPlatformVersion(result)
+            "requestPermission" -> requestPermission()
+            "checkPermission" -> checkPermission()
+            else -> result.notImplemented()
+        }
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = binding.activity
+        binding.addActivityResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        TODO("Not yet implemented")
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onDetachedFromActivity() {
+        TODO("Not yet implemented")
+    }
+
+    private fun getPlatformVersion(result: Result) {
+        result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    }
+
+    private fun requestPermission() {
+        context?.run {
+            activity?.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    500
```

Comment:
This request code should be a constant, beacuse if it is not, it is considered a magic number

Reply:
done